solr/solrj/src/java/org/apache/solr/common/util/GlobPatternUtil.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.common.util;
+
+import java.nio.file.FileSystems;
+import java.nio.file.Paths;
+
+public class GlobPatternUtil {
+
+  public static boolean matches(String pattern, String input) {
+    return FileSystems.getDefault().getPathMatcher("glob:" + pattern).matches(Paths.get(input));
```

Comment:
This is bizarre just for some filename wildcard support.  If there is a reason we use FileSystems then a comment is necessary.

Reply:
I took a look at how it's implemented.  If only we could call ZipUtils.toRegexPattern but the class is package protected.  It's a shame to recompile the glob on each call to matches!